derp/derp_client.go

```diff
@@ -559,6 +561,9 @@ func (c *Client) recvTimeout(timeout time.Duration) (m ReceivedMessage, err erro
 					binary.BigEndian.Uint16(b[keyLen+16:keyLen+16+2]),
 				)
 			}
+			if n >= keyLen+16+2+1 {
```

Comment:
And a comment here and above about how this works

Reply:
rewrote this to be obvious